Wikipedia/Code/WebViewController.m

```diff
@@ -147,6 +147,30 @@ - (void)viewWillAppear:(BOOL)animated {
                                                object:nil];
     // should happen in will appear to prevent bar from being incorrect during transitions
     [self updateZeroState];
+
+    @weakify(self);
+    void (^ saveOpenArticleTitleBlock)(NSNotification*) = ^void (NSNotification* notification) {
+        @strongify(self);
+        if (self.navigationController.topViewController == self.parentViewController) { // Ensure only the topmost article is recorded.
+            [self saveOpenArticleTitleWithCurrentlyOnscreenFragment];
+        }
+    };
+
+    [[NSNotificationCenter defaultCenter] addObserverForName:UIApplicationWillResignActiveNotification
```

Comment:
@montehurd you have to retain the object that is returned by `addObserverForName`. But honestly I would just use the normal selector API instead. The block based API is pretty wonky.

Sorry I didn't catch this earlier.

Once you make that change this is good to go.